### PR TITLE
refactor: Introduce `LogAcceptDebug()` which is `LogAcceptCategory()` with `Debug` level

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
#6399 introduced severity-based logging via b046e091c91c8b90f7f1370203d4f1d2f85c4537. We use `LogAcceptCategory()` in quite a few places and it's always `BCLog::Level::Debug` so  log level is kind of redundant for us and just makes it harder to read the code.

## What was done?
~Set log level in `LogAcceptCategory()` to `BCLog::Level::Debug` by default~. Introduce `LogAcceptDebug()` which is `LogAcceptCategory()` with `Debug` level. Simplify corresponding Dash-specific code.

## Breaking Changes
n/a

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/logging.h [hunk 1]
```diff
@@ -226,7 +226,7 @@ namespace BCLog {
 BCLog::Logger& LogInstance();
 
 /** Return true if log accepts specified category, at the specified level. */
-static inline bool LogAcceptCategory(BCLog::LogFlags category, BCLog::Level level)
+static inline bool LogAcceptCategory(BCLog::LogFlags category, BCLog::Level level = BCLog::Level::Debug)
```
Comment (knst) [inline src/logging.h:229]: consider alternate solution: e1873d7c70191d3b9fff2efa4f2f410819e927a1

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -3432,11 +3432,11 @@ std::vector<CompactTallyItem> CWallet::SelectCoinsGroupedByAddresses(bool fSkipD
     }
 
     // debug
-    if (LogAcceptCategory(BCLog::SELECTCOINS, BCLog::Level::Debug)) {
+    if (LogAcceptCategory(BCLog::SELECTCOINS)) {
         std::string strMessage = "SelectCoinsGroupedByAddresses - vecTallyRet:\n";
         for (const auto& item : vecTallyRet)
             strMessage += strprintf("  %s %f\n", EncodeDestination(item.txdest), float(item.nAmount)/COIN);
-        LogPrint(BCLog::SELECTCOINS, "%s", strMessage); /* Continued */
+        LogPrintf("%s", strMessage); /* Continued */
```
Comment (knst) [inline src/wallet/wallet.cpp:3439]: this one change behavior: you lost category from the log.

```
 node0 2024-11-29T09:38:35.061777Z (mocktime: 2014-12-04T17:15:49Z) [httpworker.0] [wallet/coinselection.cpp:279] [KnapsackSolver] [selectcoins] tryDenom: 0 
```
vs
```
 node0 2024-11-29T09:38:35.061777Z (mocktime: 2014-12-04T17:15:49Z) [httpworker.0] [wallet/coinselection.cpp:279] [KnapsackSolver] tryDenom: 0 
```

File: src/wallet/wallet.cpp [hunk 3]
```diff
@@ -3432,11 +3432,11 @@ std::vector<CompactTallyItem> CWallet::SelectCoinsGroupedByAddresses(bool fSkipD
     }
 
     // debug
-    if (LogAcceptCategory(BCLog::SELECTCOINS, BCLog::Level::Debug)) {
+    if (LogAcceptDebug(BCLog::SELECTCOINS)) {
```
Comment (knst) [inline src/wallet/wallet.cpp:3435]: this one is bitcoin's original code https://github.com/bitcoin/bitcoin/blob/master/src/wallet/coinselection.cpp#L736

Comment (UdjinM6) [reply src/wallet/wallet.cpp:3435]: That's a different function introduced in https://github.com/dashpay/dash/pull/906

**Review by knst — DISMISSED** [review]
utACK c62b35c7aec0a1d9338b817885414c8dfb51002d

**Review by kwvg — DISMISSED** [review]
utACK c62b35c7aec0a1d9338b817885414c8dfb51002d

**Comment by UdjinM6** [comment]
NOTE: we should merge #6441 first to avoid backport conflicts (will rebase this PR later)

**Comment by PastaPastaPasta** [comment]
Marked as draft so I don't merge

**Comment by PastaPastaPasta** [comment]
Range diff looks good; gonna merge

**Review by kwvg — APPROVED** [review]
utACK dfe86b4fb21199b13f7263d7c8dd0f968dd36e15

**Pull request merged**